src/shady-css/token-cursor.js

```diff
@@ -0,0 +1,36 @@
+const index = Symbol('index');
+const lexer = Symbol('lexer');
+const nextToken = Symbol('nextToken');
+
+class TokenCursor {
```

Comment:
I looked around at the `takeOne()` uses and I don't think you need it. If you separate the tokenizer current value from the act of advancing to the next token, then you can always look at `Tokenizer.currentToken` before deciding to call `Tokenizer.advance()`. The parse functions can then assume that `currentToken` is the first token that they should look at.